### ISSUE TITLE
[Bug]: obs_as_tensor RuntimeError with numpy==2.1.1 on MacOS13

### ISSUE THREAD

**Issue by nquetschlich** [DLR-RM/stable-baselines3#2016]
### 🐛 Bug

I used `sb3` in a rather comprehensive project and get a `RuntimeError: Could not infer dtype of numpy.int64` under MacOS13 when using numpy>v2.0.0. 
I boiled the problematic part down to the small example below that throws the error under the described setup. When using numpy <v2, this is not a problem.

### To Reproduce

```python
from stable_baselines3.common.utils import obs_as_tensor
from numpy import array

obs = OrderedDict([("a", array([7]))])
obs_as_tensor(device="cpu", obs=obs)
```

### Relevant log output / Error message

```shell
______________________________ test_obs_as_tensor ______________________________
    def test_obs_as_tensor() -> None:
        """Test the conversion of an observation to a tensor."""
        obs = OrderedDict([("a", array([7]))])
>       obs_as_tensor(device="cpu", obs=obs)
obs        = OrderedDict([('a', array([7]))])
tests/compilation/test_helper_rl.py:23: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/stable_baselines3/common/utils.py:487: in obs_as_tensor
    return {key: th.as_tensor(_obs, device=device) for (key, _obs) in obs.items()}
        device     = 'cpu'
        obs        = OrderedDict([('a', array([7]))])
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
.0 = <odict_iterator object at 0x136887470>
>   return {key: th.as_tensor(_obs, device=device) for (key, _obs) in obs.items()}
E   RuntimeError: Could not infer dtype of numpy.int64
.0         = <odict_iterator object at 0x136887470>
_obs       = array([7])
device     = 'cpu'
key        = 'a'
/Library/Frameworks/Python.framework/Versions/3.11/lib/python3.11/site-packages/stable_baselines3/common/utils.py:487: RuntimeError
```

### System Info

MacOS 13
Python 3.11
numpy 2.1.1
stable_baselines3 2.3.2

### Checklist

- [X] My issue does not relate to a custom gym environment. (Use the custom gym env template instead)
- [X] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo
- [X] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/)
- [X] I have provided a [minimal and working](https://github.com/DLR-RM/stable-baselines3/issues/982#issuecomment-1197044014) example to reproduce the bug
- [X] I've used the [markdown code blocks](https://help.github.com/en/articles/creating-and-highlighting-code-blocks) for both code and stack traces.

**Comment by araffin** [comment]
Probably duplicate of https://github.com/DLR-RM/stable-baselines3/issues/1960

**Comment by nquetschlich** [comment]
> Probably duplicate of #1960

I investigated further: The problem for us was, that torch added some fixes in v2.4.1 to be compatible with numpy v2 but since v2.3 macOS x86 is not supported anymore.
